VulkanHppGenerator.cpp

```diff
@@ -352,6 +352,48 @@ const std::string arrayProxyHeader = R"(
 #endif
 )";
 
+const std::string structureChainHeader = R"(
+  template <typename X, typename Y> constexpr bool isStructureChainValid() { return false; }
+
+  template <class Element>
+  class StructureChainElement
+  {
+  public:
+      operator Element&() { return value; }
+      operator const Element&() const { return value; }
+  private:
+      Element value;
+  };
+
+  template<typename ...StructureElements>
+  class StructureChain : private StructureChainElement<StructureElements>...
+  {
+  public:
+    StructureChain()
+    {
+      link<StructureElements...>();  
+    }
+
+    template<typename ClassType> ClassType& get() { return static_cast<ClassType&>(StructureChainElement<ClassType>());}
+
+  private:
+    template<typename X>
+    void link()
+    {
+    }
+
+    template<typename X, typename Y, typename ...Z>
+    void link()
+    {
+      static_assert(isStructureChainValid<X,Y>(), "The structure chain is not valid!");
+      X& x = *this;
+      Y& y = *this;
```

Comment:
Maybe some explicit cast here, even though they might not be needed. Just for the human reader.